From f814446e9032d64f34f2b02e4f8b899db31688e6 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marcos=20Chavarr=C3=ADa=20Teijeiro?=
 <chavarria1991@gmail.com>
Date: Mon, 17 Nov 2014 10:38:03 +0100
Subject: Port HTML viewer to WebKit2

I have updated the configure.ac dependencies and change viewer
implementation. The enable-default-context-menu setting doesn't exists
so I connected the context-menu signal to a functions that returns
true to do the same.

https://bugzilla.gnome.org/show_bug.cgi?id=740242
---
 configure.ac           |  2 +-
 src/js/viewers/html.js | 16 +++++++---------
 2 files changed, 8 insertions(+), 10 deletions(-)

diff --git a/configure.ac b/configure.ac
index c2f7359..0f9f741 100644
--- a/configure.ac
+++ b/configure.ac
@@ -72,7 +72,7 @@ PKG_CHECK_MODULES(SUSHI,
                   evince-document-3.0
                   evince-view-3.0
                   gtksourceview-3.0
-                  webkitgtk-3.0)
+                  webkit2gtk-4.0)
 
 GLIB_MKENUMS=`$PKG_CONFIG --variable=glib_mkenums glib-2.0`
 AC_SUBST(GLIB_MKENUMS)
diff --git a/src/js/viewers/html.js b/src/js/viewers/html.js
index 8cd9aa8..ae6bb6c 100644
--- a/src/js/viewers/html.js
+++ b/src/js/viewers/html.js
@@ -26,7 +26,7 @@
 const GtkClutter = imports.gi.GtkClutter;
 const Gtk = imports.gi.Gtk;
 const GLib = imports.gi.GLib;
-const WebKit = imports.gi.WebKit;
+const WebKit = imports.gi.WebKit2;
 const Sushi = imports.gi.Sushi;
 
 const MimeHandler = imports.ui.mimeHandler;
@@ -47,18 +47,16 @@ HTMLRenderer.prototype = {
         this._file = file;
         this._callback = callback;
 
-        this._webView = WebKit.WebView.new();
-        this._scrolledWin = Gtk.ScrolledWindow.new (null, null);
-        this._scrolledWin.add(this._webView);
-        this._scrolledWin.show_all();
+        this._webView = new WebKit.WebView();
+        this._webView.show_all();
 
         /* disable the default context menu of the web view */
-        let settings = this._webView.settings;
-        settings.enable_default_context_menu = false;
+        this._webView.connect ("context-menu",
+                               function() {return true;});
 
         this._webView.load_uri(file.get_uri());
 
-        this._actor = new GtkClutter.Actor({ contents: this._scrolledWin });
+        this._actor = new GtkClutter.Actor({ contents: this._webView });
         this._actor.set_reactive(true);
 
         this._callback();
@@ -92,7 +90,7 @@ HTMLRenderer.prototype = {
 
         return this._toolbarActor;
     }
-}
+};
 
 let handler = new MimeHandler.MimeHandler();
 let renderer = new HTMLRenderer();
-- 
cgit v0.12

